### PR TITLE
Crash fix

### PR DESCRIPTION
**Pull request by pkalever**
* fill errMsg and delete metafile on EntryCreate fail
* fix other compiler warnings

### CODE REVIEW

**Comment by pkalever** [comment]
@pranithk @lxbsz please review.

Thanks!

File: rpc/block_svc_routines.c [hunk 1]
```diff
@@ -2591,7 +2593,7 @@ getTpgObj(char *block, MetaInfo *info, blockGenConfigCli *blk, char *portal, int
   uuid_generate(uuid);
   uuid_unparse(uuid, alias);
   snprintf(alias_10, 11, "%.10s", alias+24);
-  json_object_object_add(tpg_lun_obj, "alias", GB_JSON_OBJ_TO_STR(alias_10));
+  json_object_object_add(tpg_lun_obj, "alias", GB_JSON_OBJ_TO_STR(alias_10[0]?alias_10:NULL));
```
Comment (lxbsz) [inline rpc/block_svc_routines.c]: Are this and the following changes really needed here ?

**Comment by pranithk** [comment]
@pkalever Could you add details about what was the problem explaining the reason for crash and a little bit more about the fix to the commit message?

File: rpc/block_svc_routines.c (hunk 1 above)
Comment (pkalever) [reply rpc/block_svc_routines.c]: These are to avoid gcc and rpmbuild warnings. Are you saying we should make this as a separate patch/PR ?

**Comment by pkalever** [comment]
@pranithk Sure.

**Comment by pkalever** [comment]
@pranithk @lxbsz updated the PR. Thanks!

File: rpc/block_svc_routines.c (hunk 1 above)
Comment (lxbsz) [reply rpc/block_svc_routines.c]: I test the old code, couldn't see any warning info of this. I thought the alias_10[0] couldn't be '\0' because of the last line of the code would fill something in it.

Comment (pkalever) [reply rpc/block_svc_routines.c]: Nice. Could you try:

$ make clean
$ make CFLAGS="-Wall"

Thanks!

Comment (lxbsz) [reply rpc/block_svc_routines.c]: Cool.

**Comment by pranithk** [comment]
@pkalever glusterBlockCreateEntry() seems to already have catch-all errMsg creation:
```

 if (!errMsg) {                                                               
 GB_ASPRINTF (errMsg, "Not able to create storage for %s/%s [%s]",          
                     blk->volume, blk->block_name, strerror(*errCode));            
 }  
```

Do we know why that didn't take effect?

**Comment by lxbsz** [comment]
> @pkalever glusterBlockCreateEntry() seems to already have catch-all errMsg creation:
> 
> 
>  if (!errMsg) {                                                               
>  GB_ASPRINTF (errMsg, "Not able to create storage for %s/%s [%s]",          
>                      blk->volume, blk->block_name, strerror(*errCode));            
>  }  
> Do we know why that didn't take effect?

I think here we should check:
```
if (errMsg && !(*errMsg))
...
```

**Comment by pkalever** [comment]
@pranithk @lxbsz Good catch, I did not looked at glusterBlockCreateEntry(). I have caught this on a gdb session with the core file, just fixed it this way.

Thanks for the insight. Will fix it this way.

**Comment by pranithk** [comment]
Ah! @lxbsz good catch

**Comment by pkalever** [comment]
@pranithk @lxbsz updated it as per the suggestions.

Thanks!

**Comment by lxbsz** [comment]
@pkalever 
I was running the test  script from last night till now, haven't reproduced the crash.
And the the code looks good to me.

Thanks,

File: rpc/block_svc_routines.c [hunk 2]
```diff
@@ -3900,8 +3899,12 @@ block_create_cli_1_svc_st(blockCreateCli *blk, struct svc_req *rqstp)
   }
 
   if (glusterBlockCreateEntry(glfs, blk, gbid, &errCode, &errMsg)) {
-    LOG("mgmt", GB_LOG_ERROR, "%s volume: %s host: %s",
-        FAILED_CREATING_FILE, blk->volume, blk->block_hosts);
+    LOG("mgmt", GB_LOG_ERROR, "%s volume: %s block: %s file: %s host: %s",
+        FAILED_CREATING_FILE, blk->volume, blk->block_name, gbid, blk->block_hosts);
+    if (glusterBlockDeleteMetaFile(glfs, blk->volume, blk->block_name)) {
```
Comment (pranithk) [inline rpc/block_svc_routines.c]: Even metafile deletion is happening in the  glusterBlockCreateEntry() function? Could you also move the LOG inside that function? May be in another patch it is better to remove glfs_chdir() usage from glfs-operations.c, that way we don't have to worry about pwd ever.
